enhancements/okd-on-fedora-coreos-in-prow.md

```diff
@@ -75,6 +75,13 @@ This is a proposal to:
   - by encouraging and directing community contributors to send patches and bug fixes directly to the upstream OpenShift codebase
   - create documentation and share knowledge about OKD  
 
+### OKD branding
+
+The branding for OKD differs from OpenShift, and it is crucial to display the appropriate login page and documentation links.
+When building specific components that rely on the OKD variant of code for the correct branding, Dockerfiles must use the `TAGS`
```

Comment:
>Dockerfiles must use the `TAGS`

Why? What's the benefit over the existing solution of adding `branding` image in the payload?

Reply:
There is limited context I have here, but https://issues.redhat.com//browse/OKD-214 was filed a while ago as part of the effort to move OKD to use centos-stream and it seems like the motivation for this change was to make it so that OKD builds had the correct config by default rather than having to infer from a configmap. I am not sure what the argument against the branding image was - but speaking for this change, the default reflects the product rather than deriving it from a config which could be changed/removed

Reply:
The default image should give the correct branding, with the move to using centOS stream and doing OKD specific builds we gained the ability to do things in a different way.  and as a result we are ensuring that the default for OKD builds is to get OKD branding and for OCP to get OCP branding.

Reply:
>The default image should give the correct branding

Why? Someone uses `cluster-authentication-operator` outside of OKD?

Also, the image still contains both brandings - should the "wrong" one be removed?

Reply:
> Also, the image still contains both brandings - should the "wrong" one be removed?

by using `TAGS`, only the ocp or okd branded version will be built.

Reply:
The OKD provided bits must have OKD branding. I get that in the past due to intricacies in how we built and shipped OKD it was not an option and we did a 3rd party thing. In the past, if someone removed the branding image from their installation, they would have gotten something that looked like OCP. Apache on Fedora says it is Fedora; on CentOS and RHEL, it says it is the product that it is built as. We have to do the same. The OCP branding is masked off, and the OKD branding is masked off from OCP builds.  

Can you explain why we should not ensure the branding matches the build?

Reply:
>by using TAGS, only the ocp or okd branded version will be built.

No, it merely selects the branding type - but if OCP branding is not removed it still can be selected via a configmap. So this method doesn't prevent OKD to be rebranded into OCP using supported method. Should the OCP branding be removed then?

Reply:
> No, it merely selects the branding type - but if OCP branding is not removed it still can be selected via a configmap. So this method doesn't prevent OKD to be rebranded into OCP using supported method. Should the OCP branding be removed then?

which config map is this? is it the console-config in the openshift-console namespace? even so,  the argument here is for controlling the default and it is fine if users want to change configmap values to rebrand if they choose to.

Reply:
[This one](https://github.com/openshift/origin-branding/blob/master/manifests/0000_10_origin-branding_configmap.yaml) in openshift-config-managed.

>it is fine if users want to change configmap values

No, Dennis insists [here](https://github.com/openshift/enhancements/pull/1672#discussion_r1750792734) it _must_ have OKD branding

Reply:
I said we must use OKD branding by default. just like we can not stop someone using Fedora and intentionally switching the branding to RHEL we can not stop someone switching the OKD branding to OCP.

Reply:
>I said we must use OKD branding by default

We already have a mechanism to do that, why introduce a new one?

Reply:
It is not a new mechanism; it been around for a while, and we are documenting it. We are documenting the existing mechanism. The old branding image is extra work that we will not do anymore.  The branding image was also an undocumented way to manage branding.